cli/lib/kontena/cli/stacks/yaml/stack_file_loader.rb

```diff
@@ -0,0 +1,95 @@
+require_relative '../stack_name'
+require 'yaml'
+require_relative 'reader'
+
+module Kontena::Cli::Stacks
+  module YAML
+    class StackFileLoader
+      # A base class for loading stack files. You can define more loaders by
+      # inheriting from this class.
+      #
+      # The purpose of StackFileLoader is to provide a generic interface for
+      # loading stack YAML's from different sources, such as local files,
+      # stack registry or URLs
+
+      def self.inherited(where)
+        loaders << where
+      end
+
+      def self.loaders
+        @loaders ||= []
+      end
+
+      # The main interface for getting a new loader
+      #
+      # @param source [String] stack file source string (filename, url, ..)
+      # @param parent [StackFileLoader] define a parent for recursion
+      # @return [StackFileLoader]
+      def self.for(source, parent = nil)
+        loader = loaders.find { |l| l.match?(source, parent) }
+        raise "Can't determine stack file origin for '#{source}'" if loader.nil?
+        loader.new(source, parent)
+      end
+
+      attr_reader :source, :parent
+
+      # @param source [String] stack file source string (filename, url, ..)
+      # @param parent [StackFileLoader] define a parent for recursion
+      # @return [StackFileLoader]
+      def initialize(source, parent = nil)
+        @source = source
+        @parent = parent
+      end
+
+      # @return [Hash] a hash parsed from the YAML content
+      def yaml
+        @yaml ||= reader.internals_interpolated_yaml
```

Comment:
Doesn't seem to work, I suppose the `ENV['STACK']` isnt' set at this point, so `$STACK` just interpolates to an empty string?

```
$ kontena service show deptest-bar/redis | grep PEERS
    - PEERS=foo=redis://-foo:6379/
```